pinot-core/src/main/java/org/apache/pinot/core/util/SchemaUtils.java

```diff
@@ -37,67 +41,87 @@
   public static final String MAP_KEY_COLUMN_SUFFIX = "__KEYS";
   public static final String MAP_VALUE_COLUMN_SUFFIX = "__VALUES";
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(SchemaUtils.class);
-
   /**
-   * Validates that for a field spec with transform function, the source column name and destination column name are exclusive
-   * i.e. do not allow using source column name for destination column
+   * Validates the following:
+   * 1) Checks valid transform function -
+   *   for a field spec with transform function, the source column name and destination column name are exclusive i.e. do not allow using source column name for destination column
+   *   ensure transform function string can be used to create a {@link FunctionEvaluator}
+   * 2) Checks for chained transforms/derived transform - not supported yet
+   * TODO: Transform functions have moved to table config. Once we stop supporting them in schema, remove the validations 1 and 2
+   * 3) Checks valid timeFieldSpec - if incoming and outgoing granularity spec are different a) the names cannot be same b) cannot use SIMPLE_DATE_FORMAT for conversion
+   * 4) Checks valid dateTimeFieldSpecs - checks format and granularity string
+   * 5) Schema validations from {@link Schema#validate}
    */
-  public static boolean validate(Schema schema) {
-    return validate(schema, LOGGER);
+  public static void validate(Schema schema) {
+    schema.validate();
+
+    Set<String> transformedColumns = new HashSet<>();
+    Set<String> argumentColumns = new HashSet<>();
+    for (FieldSpec fieldSpec : schema.getAllFieldSpecs()) {
+      if (!fieldSpec.isVirtualColumn()) {
+        String column = fieldSpec.getName();
+        String transformFunction = fieldSpec.getTransformFunction();
+        if (transformFunction != null) {
+          try {
+            List<String> arguments = FunctionEvaluatorFactory.getExpressionEvaluator(fieldSpec).getArguments();
+            if (arguments.contains(column)) {
+              throw new IllegalStateException("The arguments of transform function '" + transformFunction
+                  + "' should not contain the destination column '" + column + "'");
+            }
```

Comment:
For readability. Same for other checks
```suggestion
            Preconditions.checkState(arguments.contains(column), "The arguments of transform function '%s' should not contain the destination column '%s'", transformFunction, column);
```